contrib/pax_storage/Makefile

```diff
@@ -0,0 +1,132 @@
+# contrib/pax_storage/Makefile
+
+MODULE_big = pax
+OBJS = \
+	$(WIN32RES) 
+PG_CPPFLAGS = -I/usr/local/include
+PG_CXXFLAGS = -std=c++14
+
+PGFILEDESC = "pax - PAX table access method"
+SHLIB_LINK += -luuid
+
+PAX_REGRESS_DIR = src/test/regress
+PAX_VEC_REGRESS_DIR = src/test/vec
+PAX_ISOLATION2_DIR = src/test/isolation2
+
+ifdef USE_PGXS
+PG_CONFIG = pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+PGXS_DIR := $(dir $(PGXS))
+PG_REGRESS = $(PGXS_DIR)/../../src/test/regress/pg_regress
+else
+subdir = contrib/pax_storage
+top_builddir = ../../
+include $(top_builddir)/src/Makefile.global
+include $(top_srcdir)/contrib/contrib-global.mk
+PG_REGRESS = $(top_builddir)/src/test/regress/pg_regress
+endif
+
+.PHONY: all
+all: build
+
+DIRS = $(shell find src/cpp -maxdepth 2 -type d)
+SRC = $(foreach dir, $(DIRS), $(wildcard $(dir)/*.cc))
+CSRC = $(foreach dir, $(DIRS), $(wildcard $(dir)/*.c))
+
+# read options from ENV
+USE_MANIFEST_API := $(shell echo $$USE_MANIFEST_API)
+USE_PAX_CATALOG := $(shell echo $$USE_PAX_CATALOG)
+
+# if the ENVs are not set, set default value
+ifeq ($(USE_MANIFEST_API),)
+    USE_MANIFEST_API := OFF
+endif
+
+ifeq ($(USE_PAX_CATALOG),)
+    USE_PAX_CATALOG := ON
+endif
+
+.PHONY: install-data build
+build: $(SRC) $(CSRC)
+	@echo "build pax, USE_MANIFEST_API=$(USE_MANIFEST_API) USE_PAX_CATALOG=$(USE_PAX_CATALOG)"
+	@if [ ! -f build/Makefile ]; then \
+		mkdir -p build && \
+		cd build && \
+		cmake -DCMAKE_INSTALL_PREFIX=$(DESTDIR)$(prefix) \
+			-DUSE_MANIFEST_API=$(USE_MANIFEST_API) \
+			-DUSE_PAX_CATALOG=$(USE_PAX_CATALOG) .. ; \
+	fi
+	cd build && make -j8
```

Comment:
```suggestion
	cd build && make -j$(nproc)
```

Reply:
suggest not use nproc, in docker env, nproc maybe large than 8, but in fact virtual cpu cores is 4.